llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -3110,6 +3110,33 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
     }
     break;
   }
+  case ISD::STEP_VECTOR: {
+    const APInt &Step = Op.getConstantOperandAPInt(0);
+
+    if (Step.isPowerOf2())
+      Known.Zero.setLowBits(Step.logBase2());
+
+    const Function &F = getMachineFunction().getFunction();
+
+    if (!isUIntN(BitWidth, Op.getValueType().getVectorMinNumElements()))
+      break;
+    const APInt MinNumElts =
+        APInt(BitWidth, Op.getValueType().getVectorMinNumElements());
+
+    bool Overflow;
+    const APInt MaxNumElts = getVScaleRange(&F, BitWidth)
```

Comment:
Are fixed vector ISD::STEP_VECTOR allowed?

Reply:
I don't believe we construct them.  SDAGBuilder doesn't, so unless we have something somewhere which combines to them, no.

However, good point.  Unless we have an assert to that effect somewhere, we should handle them (via a bail out) here.

Reply:
Never mind, we have the assert in SelectionDAG::getNode
```
   assert(VT.isScalableVector() &&
           "STEP_VECTOR can only be used with scalable types");
    assert(OpOpcode == ISD::TargetConstant &&
           VT.getVectorElementType() == N1.getValueType() &&
           "Unexpected step operand");
```